x-pack/plugins/observability/public/components/custom_threshold/components/alert_details_app_section/alert_details_app_section.tsx

```diff
@@ -144,13 +118,14 @@ export default function AlertDetailsAppSection({
     setAlertSummaryFields(alertSummaryFields);
   }, [groups, tags, rule, ruleLink, setAlertSummaryFields]);
 
-  const derivedIndexPattern = useMemo<DataViewBase>(
-    () => ({
-      fields: dataView?.fields || [],
-      title: dataView?.getIndexPattern() || 'unknown-index',
-    }),
-    [dataView]
-  );
+  useEffect(() => {
+    let query = `${(ruleParams.searchConfiguration?.query as Query)?.query as string}`;
```

Comment:
How about having this logic in a helper function to be able to add a test for it? Didn't we already have this logic for another purpose? 🤔

Reply:
yes, I had checked code you referred to see if I can reuse, but I also found it to be different so I created logic here.

Reply:
Added helper function